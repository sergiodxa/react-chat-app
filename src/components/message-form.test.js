import React from "react";
import { render, fireEvent } from "@testing-library/react";

import MessageForm from "./message-form";

test("MessageForm component", () => {
  const ws = {
    current: {
      readyState: 1,
      send: jest.fn()
    }
  };

  const { getByLabelText } = render(
    <MessageForm isConnected channel="test" username="test-user" ws={ws} />
  );

  const input = getByLabelText("Message you want to send to the channel test");

  fireEvent.change(input, {
    target: {
      value: "Testing message"
    }
  });

  fireEvent.submit(input);

  expect(ws.current.send).toHaveBeenCalled();
});

test("MessageForm component connected", () => {
  const { asFragment } = render(
    <MessageForm
      isConnected
      channel="test"
      username="test-user"
      ws={{ current: {} }}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});

test("MessageForm component disconnected", () => {
  const { asFragment } = render(
    <MessageForm
      isConnected={false}
      channel="test"
      username="test-user"
      ws={{ current: {} }}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
