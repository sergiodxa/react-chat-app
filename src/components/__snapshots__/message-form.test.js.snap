// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm component connected 1`] = `
<DocumentFragment>
  .emotion-2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 1rem .5rem;
  font-size: 1rem;
  outline: none;
  border: none;
  border-top: 1px solid #ccc;
}

.emotion-0:disabled {
  background-color: #eaeaea;
}

.emotion-1 {
  background: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-top: 1px solid #ccc;
  padding: 1rem 1rem;
  font-size: 1rem;
  outline: none;
  -webkit-transition: color 300ms ease;
  transition: color 300ms ease;
  color: #666;
}

.emotion-1:disabled {
  background-color: #eaeaea;
}

<form
    class="emotion-2"
  >
    <input
      aria-label="Message you want to send to the channel test"
      autocomplete="false"
      class="emotion-0"
      placeholder="Message #test"
      type="text"
      value=""
    />
    <button
      aria-label="Send message"
      class="emotion-1"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
        />
      </svg>
    </button>
  </form>
</DocumentFragment>
`;

exports[`MessageForm component disconnected 1`] = `
<DocumentFragment>
  .emotion-2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 1rem .5rem;
  font-size: 1rem;
  outline: none;
  border: none;
  border-top: 1px solid #ccc;
}

.emotion-0:disabled {
  background-color: #eaeaea;
}

.emotion-1 {
  background: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  border-top: 1px solid #ccc;
  padding: 1rem 1rem;
  font-size: 1rem;
  outline: none;
  -webkit-transition: color 300ms ease;
  transition: color 300ms ease;
  color: #666;
}

.emotion-1:disabled {
  background-color: #eaeaea;
}

<form
    class="emotion-2"
    disabled=""
  >
    <input
      aria-label="Message you want to send to the channel test"
      autocomplete="false"
      class="emotion-0"
      disabled=""
      placeholder="Message #test"
      type="text"
      value=""
    />
    <button
      aria-label="Send message"
      class="emotion-1"
      disabled=""
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
        />
      </svg>
    </button>
  </form>
</DocumentFragment>
`;
